Spec/Auth.swift

```diff
@@ -637,9 +637,115 @@ class Auth : QuickSpec {
+
+                it("query will provide a token string") {
+                    let testToken = getTestToken()
+
+                    let options = ARTClientOptions()
+                    options.authUrl = NSURL(string: "http://echo.ably.io")
+                    expect(options.authUrl).toNot(beNil())
+                    // Plain text
+                    options.authParams = [NSURLQueryItem]()
+                    options.authParams!.append(NSURLQueryItem(name: "type", value: "text"))
+                    options.authParams!.append(NSURLQueryItem(name: "body", value: testToken))
+
+                    let rest = ARTRest(options: options)
+                    rest.httpExecutor = testHTTPExecutor
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.requestToken(nil, withOptions: nil, callback: { tokenDetails, error in
+                            expect(testHTTPExecutor.requests.last?.URL?.host).to(equal("echo.ably.io"))
+                            expect(error).to(beNil())
+                            expect(tokenDetails).toNot(beNil())
+                            expect(tokenDetails?.token).to(equal(testToken))
+                            done()
+                        })
+                    }
+                }
+
+                pending("query will provide a TokenDetails") {
+                    guard let testTokenDetails = getTestTokenDetails() else {
+                        fail("TokenDetails is empty")
+                        return
+                    }
+
+                    let encoder = ARTJsonLikeEncoder()
+                    encoder.delegate = ARTJsonEncoder()
+                    guard let jsonTokenDetails = encoder.encodeTokenDetails(testTokenDetails) else {
+                        fail("Invalid TokenDetails")
+                        return
+                    }
+
+                    let options = ARTClientOptions()
+                    options.authUrl = NSURL(string: "http://echo.ably.io")
+                    expect(options.authUrl).toNot(beNil())
+                    // JSON with TokenDetails
+                    options.authParams = [NSURLQueryItem]()
+                    options.authParams?.append(NSURLQueryItem(name: "type", value: "json"))
+                    options.authParams?.append(NSURLQueryItem(name: "body", value: jsonTokenDetails.toUTF8String))
+
+                    let rest = ARTRest(options: options)
+                    rest.httpExecutor = testHTTPExecutor
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.requestToken(nil, withOptions: nil, callback: { tokenDetails, error in
+                            expect(testHTTPExecutor.requests.last?.URL?.host).to(equal("echo.ably.io"))
+                            expect(error).to(beNil())
+                            expect(tokenDetails).toNot(beNil())
+                            expect(tokenDetails).to(equal(testTokenDetails))
```

Comment:
**Pending because** `tokenDetails.issued` is nil and `tokenDetails.expires` is nil.

But `jsonTokenDetails.toUTF8String` is `"http://echo.ably.io?ttl=3600.000000&capability=%7B%22*%22:%5B%22*%22%5D%7D&timestamp=1472479389.001963&type=json&body=%7B%22expires%22:%221472482988000%22,%22capability%22:%22%7B%5C%22*%5C%22:%5B%5C%22*%5C%22%5D%7D%22,%22token%22:%22WKZpMg.Dcg4-NM2v5eYV5v8bmkTRxv4oe3nmp7qmQ4DA-5lXij4tcizw7uIex0vMrOtNUqRcSZhGR9z9d5H60dZBayVTpcABAO80d4K9_F6I-IgpMryU7md31s9IbmPkrZvuU7Rs%22,%22issued%22:%221472479388000%22%7D&format=msgpack"`

Both `issued` and `expires` have values and `echo.ably.io` is working fine.

```
curl -X "GET" "http://echo.ably.io/?type=json&body=%7B%22expires%22:%221472482988000%22,%22capability%22:%22%7B%5C%22*%5C%22:%5B%5C%22*%5C%22%5D%7D%22,%22token%22:%22WKZpMg.Dcg4-NM2v5eYV5v8bmkTRxv4oe3nmp7qmQ4DA-5lXij4tcizw7uIex0vMrOtNUqRcSZhGR9z9d5H60dZBayVTpcABAO80d4K9_F6I-IgpMryU7md31s9IbmPkrZvuU7Rs%22,%22issued%22:%221472479388000%22%7D"
```

cURL returns:

```
{"expires":"1472482988000","capability":"{\"*\":[\"*\"]}","token":"WKZpMg.Dcg4-NM2v5eYV5v8bmkTRxv4oe3nmp7qmQ4DA-5lXij4tcizw7uIex0vMrOtNUqRcSZhGR9z9d5H60dZBayVTpcABAO80d4K9_F6I-IgpMryU7md31s9IbmPkrZvuU7Rs","issued":"1472479388000"}
```